doc/ref_model/chapters/chapter04.md

```diff
@@ -262,6 +263,32 @@ Table 4-10 shows performance measurement capabilities.
+
+**Table 4-12** shows Acceleration/Offload API Capabilities.
+
+<a name="Table4-12"></a>
+
+| Ref       | Acceleration/Offload API Capability | Unit   | Definition/Notes                                                                            |
+|-----------|--------------------------------------------|--------|---------------------------------------------------------------------------------------------|
+| e.api.001 | VNF/CNF usage of Accelerator standard i/f                           | Yes/No | VNF/CNF shall use abstracted standardized interfaces to the Acceleration/Offload function that enables multiple HW and SW implementation of the accelerated/offloaded function from multiple vendors in the Cloud Infrastructure                        |
+| e.api.002 | Virtualization Infrastructure SW usage of Accelerator standard i/f  | Yes/No | Virtualization Infrastructure SW shall use abstracted standardized interfaces to the HW-Acceleration/Offload  function that enables multiple HW and SW implementation in the HW Infrastructure Layer of the accelerated function from multiple vendors  |
+| e.api.003 | Accelerators offering standard i/f to HW Infra Layer                | Yes/No | Acceleration/Offload functions shall offer abstracted standardized interfaces for the Virtualization Infrastructure and VNF/CNF applications                                                                                                             |
+| e.api.004 | Accelerators offering virtualized functions                         | Yes/No | Acceleration/Offload functions for VNFs/CNFs should preferably be virtualized to allow multiple VNFs/CNFs to use the same Acceleration/Offload instance                                                                                                 |
+| e.api.005 | VNF/CNF Accelerator management functions access rights              | Yes/No | VNF/CNF management functions shall be capable to request Acceleration/Offload invocation without having elevated access rights                                                                                                                       |
+| e.api.006 | Accelerators offering standard i/f to VNF/CNF management            | Yes/No | VNF/CNF management functions should be capable to request Acceleration/Offload invocation through abstracted standardized Management interfaces                                                                                                       |
```

Comment:
Would this mean Cyborg in OpenStack and standardised API schemes for Device Plugins in Kubernetes?

Reply:
To my knowledge (but I am not an expert here) Cyborg is most a discovery and mapping interface for some resource orchestrator, and that is a possible API that could make sense in the management API area of the technology dependent RA-1.
The Device Plug-In I would think (again, I am no expert in this area) can abstract underlying implementation differences, so a specification of what a Device Plug-In can do and how to interface/manage Device Plug-Ins would be an example of an architecture dependent API that should be standardized in RA-2.

Reply:
@TFredberg @CsatariGergely Again not an expert but Cyborg supports LCM for accelerators. It supports a subset of compute (Nova) operations for servers (in OSTK -- physical, VM or container) with accelerators.  So as to avoid flavor explosion, Cyborg uses the concept of device profiles -- a sort of "flavor" for devices -- that can be included in regular OSTK flavor definitions. Another distinction is an [abstraction of drivers](https://docs.openstack.org/cyborg/latest/user/architecture.html) which then interacts with specific vendor drivers -- so an abstract driver for FPGA, another for GPU, etc.

Reply:
In my reading e.api.006 is about the management of accelerators and not the consumption of them. I also agree with @pgoyal01 in that Cyborg is more than a simple discovery and mapping interface. I can reach out for the Cyborg community in OpenStack to have a more experienced opinion. 
Device plugins are already [required in RA2](https://cntt.readthedocs.io/en/latest/ref_arch/kubernetes/chapters/chapter03.html?highlight=device%20plugin#device-plugin-framework). The issue with device plugins is that every device plugin offers a different API, so "device plugin" does not mean a standard interface to manage accelerators, it means a standard approach. If we would like to have a standard api we would need to specify a scheme for it and influence device plugins to use the scheme.

Reply:
The intent of e.api.006 is to have a decoupling in between the HW realization of the HW-accelerator (inside the Cloud Infrastructure) and the VNF/CNF application or its management function (like a resource orchestrator). To achieve that, there is a need to have a requirement on the Cloud Infrastructure exposed control/management interfaces. If that is fulfilled by the HW accelerator or the Virtualization Layer (OpenStack or K8s) is up to the Cloud Infrastructure realization. It is likely that the vendors of HW accelerators do not have vendor agnostic portable interfaces at least not in their early releases, and then it will be up to the Virtualization Layer to try their best to abstract the differences of the HW accelerators.

Reply:
Would agree on the discussion, but I think it will be a subject for respective RA rather than in the RM.